### PR TITLE
vmm for macOS

### PR DESCRIPTION
**Pull request by Lencerf**

### CODE REVIEW

File: xhype/xhype/src/bios.rs [hunk 1]
```diff
@@ -0,0 +1,356 @@
+use crate::mach::MachVMBlock;
```
Comment (rminnich) [inline xhype/xhype/src/bios.rs:1]: You need to put license header in each file you create. 

/* SPDX-License-Identifier: GPL-2.0-only */
will do.

Comment (Lencerf) [reply xhype/xhype/src/bios.rs:2]: license headers are added.

**Comment by Lencerf** [comment]
bios.rs, decode.rs, ioapic.rs are from Akaros.

File: xhype/xhype/LICENSE [hunk 2]
```diff
@@ -0,0 +1,340 @@
+                    GNU GENERAL PUBLIC LICENSE
```
Comment (rminnich) [inline xhype/xhype/LICENSE:1]: I am wondering why you need this here, since there is one at top level?

File: xhype/xhype/cdef2rust.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import sys
```
Comment (rminnich) [inline xhype/xhype/cdef2rust.py:1]: what license is this? where is it from? Do you need it here or can people get it some other way.

File: xhype/xhype/cstruct2rust.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
+import sys
```
Comment (rminnich) [inline xhype/xhype/cstruct2rust.py:1]: Where is this from?

File: xhype/xhype/src/hv/mod.rs [hunk 5]
```diff
@@ -0,0 +1,613 @@
+/* SPDX-License-Identifier: GPL-2.0-only */
```
Comment (rminnich) [inline xhype/xhype/src/hv/mod.rs:1]: did you write this or did you get it from somewhere else?

File: xhype/xhype/src/x86.rs [hunk 6]
```diff
@@ -0,0 +1,73 @@
+/* SPDX-License-Identifier: GPL-2.0-only */
```
Comment (rminnich) [inline xhype/xhype/src/x86.rs:1]: Can you set unused and uncomment all the commented out constants?

File: xhype/xhype/cdef2rust.py (hunk 3 above)
Comment (Lencerf) [reply xhype/xhype/cdef2rust.py:1]: Both cstruct2rust.py and cdef2rust.py are written by myself and are used for converting some C code to Rust. It is not necessary for the program. I just put here for future development. Struct in bios.rs and linux.rs and constants in msr.rs are generated by the two scripts.

File: xhype/xhype/cstruct2rust.py (hunk 4 above)
Comment (Lencerf) [reply xhype/xhype/cstruct2rust.py:1]: It is written by myself.

File: xhype/xhype/src/hv/mod.rs (hunk 5 above)
Comment (Lencerf) [reply xhype/xhype/src/hv/mod.rs:1]: The three files mod.rs, ffi.rs, and vmx.rs are based on the code in https://github.com/RWTH-OS/xhypervisor. I modified and added some other functions.

File: xhype/xhype/LICENSE (hunk 2 above)
Comment (Lencerf) [reply xhype/xhype/LICENSE:1]: It is not necessary. I can remove it since there is one at top level.

**Comment by Lencerf** [comment]
I used some codes from the crate [xhypervisor](https://crates.io/crates/xhypervisor) and some codes were generated by the python scripts from Akaros codes. I will make separate PRs.

**Pull request closed without merging**